framework/Distributions.py

```diff
@@ -1752,7 +1742,7 @@ def ppf(self,x):
     cumulative=0.0
     for element in sortedMapping:
       cumulative += element[1]
-      if cumulative >= x:
+      if cumulative >= (x - len(self.values)*np.finfo(float).eps):
```

Comment:
Could you normalize the values in self.mapping instead to add additional tolerance? 
The problem is coming from line 1701: mathUtils.compareFloats(self.totPsum,1.0), the default tolerance is 1.E-6. That's our assumption for the user input.

Reply:
I am a moron, I ddi not think on it. I have added the normalization in the uniformDiscrete (line 1865)

Reply:
Also another suggestion, it seems we do not need to compute the cumulative every time we call ppf, we can do it in the initialization method, right?